ladybug_geometry/polyskel.py

```diff
@@ -0,0 +1,558 @@
+						continue
+
+					#check eligibility of b
+					# a valid b should lie within the area limited by the edge and the bisectors of its two vertices:
+					xleft = _determinant(edge.bisector_left.v.normalized(), (b - edge.bisector_left.p).normalized()) > 0
+					xright = _determinant(edge.bisector_right.v.normalized(), (b - edge.bisector_right.p).normalized()) < 0
+					xedge = _determinant(edge.edge.v.normalized(), (b - edge.edge.p).normalized()) < 0
+
+					if not (xleft and xright and xedge):
+						log.debug("\t\tDiscarded candidate %s (%s-%s-%s)", b, xleft, xright, xedge)
+						continue
+
+					log.debug("\t\tFound valid candidate %s", b)
+					events.append( _SplitEvent(Line2(edge.edge).distance(b), b, self, edge.edge) )
+
+		# Intersect line2d with line2d (does not assume lines are infinite)
+		i_prev = intersection2d.intersect_line2d(self.bisector, self.prev.bisector)
+		i_next = intersection2d.intersect_line2d(self.bisector, self.prev.bisector)
+
+		# Calc distances from edge to intersection
+		dist_to_i_prev = LineSegment2D(
+			self.edge_left.p.duplicate(),
+			self.edge_left.v.duplicate())
+		dist_to_i_prev = dist_to_i_prev.distance_to_point(i_prev)
+		dist_to_i_next = LineSegment2D(
+			self.edge_right.p.duplicate(),
+			self.edge_right.v.duplicate())
+		dist_to_i_next = dist_to_i_next.distance_to_point(i_next)
+
+		# Make EdgeEvent and append to events
+		if i_prev is not None:
+			events.append(_EdgeEvent(dist_to_i_prev, i_prev, self.prev, self))
+		if i_next is not None:
+			events.append(_EdgeEvent(dist_to_i_next, i_next, self, self.next))
+
+		if not events:
+			return None
+
+		ev = min(
+			events,
+			key=lambda event: self.point.distance_to_point(event.intersection_point)
+			)
+
+		log.info("Generated new event for %s: %s", self, ev)
+		return ev
+
+	def invalidate(self):
+		if self.lav is not None:
+			self.lav.invalidate(self)
+		else:
+			self._valid = False
+
+	@property
+	def is_valid(self):
+		return self._valid
+
+	def __str__(self):
+		return "Vertex ({:.2f};{:.2f})".format(self.point.x, self.point.y)
+
+	def __lt__(self, other):
+		if isinstance(other, _LAVertex):
+			return self.point.x < other.point.x
+
+	def __repr__(self):
+		return "Vertex ({}) ({:.2f};{:.2f}), bisector {}, edges {} {}".format("reflex" if self.is_reflex else "convex", self.point.x, self.point.y, self.bisector, self.edge_left, self.edge_right)
+
+class _SLAV:
+	def __init__(self, polygon, holes):
+		contours = [_normalize_contour(polygon)]
+		contours.extend([_normalize_contour(hole) for hole in holes])
+
+		self._lavs = [ _LAV.from_polygon(contour, self) for contour in contours ]
+
+		#store original polygon edges for calculating split events
+		self._original_edges = [
+			_OriginalEdge(
+				LineSegment2D.from_end_points(vertex.prev.point, vertex.point),
+				vertex.prev.bisector,
+				vertex.bisector
+				) for vertex in chain.from_iterable(self._lavs)
+			]
+
+	def __iter__(self):
+		for lav in self._lavs:
+			yield lav
```

Comment:
what about `return iter(self._lavs)`?

Reply:
Is iter() return a generator? If so it should be equivalent.

Reply:
So I looked this up and generators are a simpler type of iterators that should be a little faster: https://stackoverflow.com/questions/2776829/difference-between-pythons-generators-and-iterators

Unless anyone has any objections, I'll keep as is.